### PR TITLE
Rework DeviceStatusSensor with SimpleAggregateSensor

### PR DESCRIPTION
**Pull request by bmerry**
This simplifies the logic somewhat, and makes the timestamp logic the responsibility of aiokatcp. A functional change is that now if the worst status is beyond ERROR (e.g. UNREACHABLE), the aggregate status will be ERROR. That shouldn't have any effect since katgpucbf does not provide any such sensors.

This will fail to run right now because it needs a new release of aiokatcp that implements SimpleAggregateSensor.

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] If dependencies are added/removed: update ``setup.cfg`` and appropriate requirements files
- [x] (N/A) If modules are added/removed: use sphinx-apidoc to update files in ``doc/``
- [x] Ensure copyright notices are present and up-to-date
- [x] (N/A) If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date

Relates to NGC-442.

### CODE REVIEW

File: src/katgpucbf/utils.py
Comment (james-smith-za) [inline src/katgpucbf/utils.py:80]: I did vaguely think of keeping lists / counters / some kind of track of sensors in various states. It seemed more complicated to do it this way though, not less.

**Review by james-smith-za — APPROVED** [review]
I don't personally feel as though this simplifies things substantially over what I had. You don't have to handle the timestamp, but you need to add two extra functions, so it ends up being a similar number of lines.

But I don't feel strongly enough about it either way to block it.

**Comment by bmerry** [comment]
> You don't have to handle the timestamp, but you need to add two extra functions, so it ends up being a similar number of lines.

The functions are very simple and easy to follow though - you no longer need to think about what it's going to happen in each of the 4 different cases. So even if LOC stays the same I think the cyclomatic complexity is most likely reduced.

> I did vaguely think of keeping lists / counters / some kind of track of sensors in various states. It seemed more complicated to do it this way though, not less.

It's more scalable though: each sensor update takes O(1) time to process now, rather than O(N).

**Comment by james-smith-za** [comment]
> 
> It's more scalable though: each sensor update takes O(1) time to process now, rather than O(N).

This is a fair point.

File: src/katgpucbf/utils.py
Comment (amishatishpatel) [inline src/katgpucbf/utils.py:106]: This is very cool.

**Review by amishatishpatel — APPROVED** [review]

**Comment by bmerry** [comment]
Have now bumped the aiokatcp dependency to 1.5.0. I've requested re-reviews from both of you to double-check that I didn't miss anything, but I reckon I'll merge once either of you (rather than both) gets around to it rather than waiting for both.

**Comment by james-smith-za** [comment]
> Have now bumped the aiokatcp dependency to 1.5.0. I've requested re-reviews from both of you to double-check that I didn't miss anything, but I reckon I'll merge once either of you (rather than both) gets around to it rather than waiting for both.

Just waiting for Jenkins to finish before I click approve.

**Comment by bmerry** [comment]
> Just waiting for Jenkins to finish before I click approve.

I'll take that as permission to merge if Jenkins is happy.

**Review by james-smith-za — APPROVED** [review]

**Pull request merged**
